### ISSUE TITLE
DevTools HTTP Request Failed Error with Chromedriver and chrome not reachable

### ISSUE THREAD

**Issue by omarzero130** [appium/appium#20395]
### Do I have the most recent component updates?

- [X] I use the most recent available driver/plugin and server versions

### Is the component officially supported by the Appium team?

- [X] I have verified the component repository is present under the Appium organization in GitHub

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Currently iam trying to run my webview app on device farm locally it runs fine but on device farm the app opens but the tests are not executed and it stuck on the app screen without running the tests and the following error occurs repeatedly, causing the test to fail:

`[Chromedriver@3244] [STDERR] [1721226339.409][DEBUG]: DevTools HTTP Request failed
[Chromedriver@3244] [STDERR] [1721226339.459][DEBUG]: DevTools HTTP Request: http://localhost:37869/json/version
[Chromedriver@3244] [STDERR] [1721226339.460][DEBUG]: DevTools HTTP Request failed
Could not start a new session. Response code 500. Message: An unknown server-side error occurred while processing the command. Original error: A new session could not be created. Details: session not created
from chrome not reachable 
`

### Expected Behavior

Chromedriver should start without any errors and the tests should run successfully.

### Minimal Reproducible Example

The following attached code is my appium intializing 

`public static AppiumDriver intialize() throws MalformedURLException {

        WebDriverManager.chromedriver().setup();

        String chromedriverPath = WebDriverManager.chromedriver().getDownloadedDriverPath();
 
        UiAutomator2Options options = new UiAutomator2Options();
        options
                .setApp(JsonReader.getJsonValue("capabilities.appPath"))
                .setPlatformName(JsonReader.getJsonValue("capabilities.platformName"))
                .setPlatformVersion(JsonReader.getJsonValue("capabilities.platformVersion"))
                .setChromedriverPort(9515)
                .setShowChromedriverLog(true)
                .autoGrantPermissions()
                .disableWindowAnimation()
                .setChromedriverExecutable(chromedriverPath)
                .setAutoWebview(true);
        System.out.println("UiAutomator2Options is +++++++++++++++++++++++++++++++++++: " + options);
        appiumDriver = new AndroidDriver(new URL(APPIUM_SERVER_URL), options);
        appiumDriver.setFileDetector(new LocalFileDetector());
        System.out.println("Current Context: " + ((SupportsContextSwitching) appiumDriver).getContext());
        return appiumDriver;

    }`

### Environment

AWS Device Farm: Using Amazon Linux 2
Node.js: 18.x
Appium: 2.5.1
Chromedriver: Latest version installed using WebDriverManager(io.github.bonigarcia)
Android SDK Tools: Latest version
Device : Google pixel 4 a

**Comment by mykola-mokhnach** [comment]
- Please use most recent driver version
- Please provide the full server log

**Comment by mykola-mokhnach** [comment]
I assume it's an issue of the device farm or your test configuration. Try contacting their customer support

**Comment by omarzero130** [comment]
@mykola-mokhnach  i am using the following web driver manager https://mvnrepository.com/artifact/io.github.bonigarcia/webdrivermanager/5.8.0  to handle the chrome driver if the device is running uses old chrome version 

The full log is bit large do you want to provide you with a certain part of the logs and here is some parts

`[38;5;32m[AndroidUiautomator2Driver@de6a][0m Passing web view details to the Chromedriver constructor: {
  "info": {
    "Android-Package": "com.test.stage",
    "Browser": "Chrome/126.0.6478.133",
    "Protocol-Version": "1.3",
    "User-Agent": "Mozilla/5.0 (Linux; Android 11; Pixel 4a Build/RQ3A.210605.005; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/126.0.6478.133 Mobile Safari/537.36",
    "V8-Version": "12.6.228.21",
    "WebKit-Version": "537.36 (@b8ff67a29a292f74521f9726bbb118457080b9a1)",
    "webSocketDebuggerUrl": "ws://127.0.0.1:10900/devtools/browser"
  },
  "process": {
    "name": "com.test.stage",
    "id": "4725"
  }
}
[38;5;32m[AndroidUiautomator2Driver@de6a][0m Precalculated Chromedriver capabilities: {
  "androidPackage": "com.test.stage",
  "androidUseRunningApp": true,
  "androidProcess": "com.test.stage",
  "androidDeviceSerial": "17291JEC207128"
}
[38;5;32m[AndroidUiautomator2Driver@de6a][0m Before starting chromedriver, androidPackage is 'com.test.stage'
[38;5;128m[Chromedriver@fc0e][0m Changed state to 'starting'
[38;5;128m[Chromedriver@fc0e][0m Using the static Chromedriver->Chrome mapping
[38;5;128m[Chromedriver@fc0e][0m The most recent known Chrome version: 123.0.6312
[38;5;128m[Chromedriver@fc0e][0m Found 2 executables in '/home/device-farm/.appium/node_modules/appium-uiautomator2-driver/node_modules/appium-chromedriver/chromedriver/linux'
[38;5;128m[Chromedriver@fc0e][0m The following Chromedriver executables were found:
[38;5;128m[Chromedriver@fc0e][0m     '/home/device-farm/.appium/node_modules/appium-uiautomator2-driver/node_modules/appium-chromedriver/chromedriver/linux/chromedriver_linux64_v123.0.6312.105' (version '123.0.6312.105', minimum Chrome version '123.0.6312')
[38;5;128m[Chromedriver@fc0e][0m     '/home/device-farm/.appium/node_modules/appium-uiautomator2-driver/node_modules/appium-chromedriver/chromedriver/linux/chromedriver-linux64_v123.0.6312.105' (version '123.0.6312.105', minimum Chrome version '123.0.6312')
[38;5;128m[Chromedriver@fc0e][0m Browser version in the supplied details: Chrome/126.0.6478.133
[38;5;128m[Chromedriver@fc0e][0m Found Chrome bundle 'undefined' version '126.0.6478'
[38;5;144m[ChromedriverGoogleapisStorageClient][0m Parsed 247 entries from storage XML
[38;5;144m[ChromedriverGoogleapisStorageClient][0m The entry '2.0/chromedriver_linux64.zip' does not contain any notes. Skipping it
[38;5;144m[ChromedriverGoogleapisStorageClient][0m The entry '2.1/chromedriver_linux64.zip' does not contain any notes. Skipping it
[38;5;144m[ChromedriverGoogleapisStorageClient][0m The entry '2.17/chromedriver_linux64.zip' does not contain any notes. Skipping it
[38;5;144m[ChromedriverGoogleapisStorageClient][0m The entry '2.2/chromedriver_linux64.zip' does not contain any notes. Skipping it
[38;5;144m[ChromedriverGoogleapisStorageClient][0m The driver '2.3/chromedriver_linux64.zip' does not contain valid release notes at https://devicefarm-prod-us-west-2-3aa76-chromedriver-stable.bucket.vpce-04b5ca87cba55a833-ojq8n2ql.s3.us-west-2.vpce.amazonaws.com/2.3/notes.txt. Skipping it
[38;5;144m[ChromedriverGoogleapisStorageClient][0m The entry '70.0.3538.97/chromedriver_linux64.zip' does not contain any notes. Skipping it
[38;5;144m[ChromedriverGoogleapisStorageClient][0m The entry '71.0.3578.137/chromedriver_linux64.zip' does not contain any notes. Skipping it
[38;5;144m[ChromedriverGoogleapisStorageClient][0m The entry '72.0.3626.69/chromedriver_linux64.zip' does not contain any notes. Skipping it
[38;5;144m[ChromedriverGoogleapisStorageClient][0m The total count of entries in the mapping: 127
[38;5;160m[ChromedriverChromelabsStorageClient][0m The total count of entries in the mapping: 3960
[38;5;128m[Chromedriver@fc0e][0m Got chromedrivers mapping from the storage: {
[38;5;128m[Chromedriver@fc0e][0m   "100.0.4896.60/chromedriver_linux64.zip": {
[38;5;128m[Chromedriver@fc0e][0m     "url": "https://devicefarm-prod-us-west-2-3aa76-chromedriver-stable.bucket.vpce-04b5ca87cba55a833-ojq8n2ql.s3.us-west-2.vpce.amazonaws.com/100.0.4896.60/chromedriver_linux64.zip",
[38;5;128m[Chromedriver@fc0e][0m     "etag": "3eb3cf16bf08657ba9cbe7f8d03f2b57",
[38;5;128m[Chromedriver@fc0e][0m     "version": "100.0.4896.60",
[38;5;128m[Chromedriver@fc0e][0m     "minBrowserVersion": "100",
[38;5;128m[Chromedriver@fc0e][0m     "os": {
[38;5;128m[Chromedriver@fc0e][0m       "name": "linux",
[38;5;128m[Chromedriver@fc0e][0m       "arch": "64",
[38;5;128m[Chromedriver@fc0e][0m       "cpu": "intel"
[38;5;128m[Chromedriver@fc0e][0m     }
[38;5;128m[Chromedriver@fc0e][0m   },
[38;5;128m[Chromedriver@fc0e][0m   "101.0.4951.41/chromedriver_linux64.zip": {
[38;5;128m[Chromedriver@fc0e][0m     "url": "https://...
[38;5;176m[ChromedriverStorageClient][0m Selecting chromedrivers whose minimum supported browser version matches to 126
[38;5;176m[ChromedriverStorageClient][0m Got 292 items
[38;5;176m[ChromedriverStorageClient][0m Will select candidate drivers versioned as '126.0.6478.126,126.0.6478.127,126.0.6478.182,126.0.6478.55,126.0.6478.61,126.0.6478.62,126.0.6478.63,126.0.6423.0,126.0.6423.2,126.0.6424.0,126.0.6436.0,126.0.6437.0,126.0.6437.2,126.0.6438.0,126.0.6438.2,126.0.6439.0,126.0.6440.0,126.0.6441.0,126.0.6442.0,126.0.6443.0,126.0.6444.0,126.0.6445.0,126.0.6446.0,126.0.6447.0,126.0.6448.0,126.0.6449.0,126.0.6456.0,126.0.6457.0,126.0.6458.0,126.0.6460.0,126.0.6461.0,126.0.6462.0,126.0.6463.0,126.0.6464.0,126.0.6465.0,126.0.6465.2,126.0.6466.0,126.0.6467.0,126.0.6467.2,126.0.6468.0,126.0.6468.2,126.0.6469.0,126.0.6469.2,126.0.6470.0,126.0.6471.0,126.0.6472.0,126.0.6473.0,126.0.6474.0,126.0.6475.0,126.0.6477.0,126.0.6477.3,126.0.6478.0,126.0.6478.2,126.0.6478.7,126.0.6478.17,126.0.6478.26,126.0.6478.36'
[38;5;176m[ChromedriverStorageClient][0m Selecting chromedrivers whose platform matches to linux:intel64
[38;5;176m[ChromedriverStorageClient][0m Got 64 items
[38;5;176m[ChromedriverStorageClient][0m Excluding older patches if present
[38;5;176m[ChromedriverStorageClient][0m Versions mapping: {
[38;5;176m[ChromedriverStorageClient][0m   "126": [
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.182",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.182",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.127",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.127",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.126",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.126",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.63",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.63",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.62",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.62",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.61",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.61",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.55",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.55",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.36",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.26",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.17",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.7",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.2",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6478.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6477.3",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6477.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6475.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6474.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6473.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6472.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6471.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6470.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6469.2",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6469.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6468.2",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6468.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6467.2",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6467.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6466.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6465.2",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6465.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6464.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6463.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6462.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6461.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6460.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6458.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6457.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6456.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6449.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6448.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6447.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6446.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6445.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6444.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6443.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6442.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6441.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6440.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6439.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6438.2",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6438.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6437.2",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6437.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6436.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6424.0",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6423.2",
[38;5;176m[ChromedriverStorageClient][0m     "126.0.6423.0"
[38;5;176m[ChromedriverStorageClient][0m   ]
[38;5;176m[ChromedriverStorageClient][0m }
[38;5;176m[ChromedriverStorageClient][0m Got 2 drivers to sync: [
[38;5;176m[ChromedriverStorageClient][0m   "126.0.6478.182/chromedriver_linux64.zip",
[38;5;176m[ChromedriverStorageClient][0m   "126.0.6478.182/chromedriver-linux64.zip"
[38;5;176m[ChromedriverStorageClient][0m ]
[38;5;176m[ChromedriverStorageClient][0m Retrieving 'https://devicefarm-prod-us-west-2-3aa76-chromedriver-stable.bucket.vpce-04b5ca87cba55a833-ojq8n2ql.s3.us-west-2.vpce.amazonaws.com/126.0.6478.182/chromedriver_linux64.zip' to '/tmp/2024618-15914-wvteme.qjxsj/0.zip'
[38;5;176m[ChromedriverStorageClient][0m Retrieving 'https://storage.googleapis.com/chrome-for-testing-public/126.0.6478.182/linux64/chromedriver-linux64.zip' to '/tmp/2024618-15914-wvteme.qjxsj/1.zip'
`

and this part before the code stucks 
`m Browser version in the supplied details: Chrome/126.0.6478.133
[38;5;128m[Chromedriver@fc0e][0m Found Chrome bundle 'undefined' version '126.0.6478'
[38;5;128m[Chromedriver@fc0e][0m Found 2 executables capable of automating Chrome '126.0.6478'.
[38;5;128m[Chromedriver@fc0e][0m Choosing the most recent, '/home/device-farm/.appium/node_modules/appium-uiautomator2-driver/node_modules/appium-chromedriver/chromedriver/linux/chromedriver_linux64_v126.0.6478.182'.
[38;5;128m[Chromedriver@fc0e][0m If a specific version is required, specify it with the `chromedriverExecutable`desired capability.
[38;5;128m[Chromedriver@fc0e][0m Set chromedriver binary as: /home/device-farm/.appium/node_modules/appium-uiautomator2-driver/node_modules/appium-chromedriver/chromedriver/linux/chromedriver_linux64_v126.0.6478.182
[38;5;128m[Chromedriver@fc0e][0m Killing any old chromedrivers, running: pkill -15 -f "/home/device-farm/.appium/node_modules/appium-uiautomator2-driver/node_modules/appium-chromedriver/chromedriver/linux/chromedriver_linux64_v126.0.6478.182.*--port=9515"
[38;5;128m[Chromedriver@fc0e][0m No old chromedrivers seem to exist
[38;5;128m[Chromedriver@fc0e][0m Cleaning this device's adb forwarded port socket connections: 17291JEC207128
[38;5;48m[ADB][0m List forwarding ports
[38;5;48m[ADB][0m Running '/tmp/devicefarm-workspace/execution-vvi7ufkc/shared-cjdq4djl/android-home-r90v3det/platform-tools/adb -P 5037 -s 17291JEC207128 forward --list'
[38;5;128m[Chromedriver@fc0e][0m Spawning chromedriver with: /home/device-farm/.appium/node_modules/appium-uiautomator2-driver/node_modules/appium-chromedriver/chromedriver/linux/chromedriver_linux64_v126.0.6478.182 --port=9515 --adb-port=5037 --verbose
[38;5;128m[Chromedriver@fc0e][0m ChromeDriver version: '126.0.6478.182'
[38;5;128m[Chromedriver@fc0e][0m [STDOUT] Starting ChromeDriver 126.0.6478.182 (5b5d8292ddf182f8b2096fa665b473b6317906d5-refs/branch-heads/6478@{#1776}) on port 9515
[38;5;128m[Chromedriver@fc0e][0m [STDOUT] Only local connections are allowed.
[38;5;128m[Chromedriver@fc0e][0m [STDOUT] Please see https://chromedriver.chromium.org/security-considerations for suggestions on keeping ChromeDriver safe.
[38;5;128m[Chromedriver@fc0e][0m Matched '/status' to command name 'getStatus'
[38;5;128m[Chromedriver@fc0e][0m Proxying [GET /status] to [GET http://127.0.0.1:9515/status] with no body
[38;5;128m[Chromedriver@fc0e][0m [STDOUT] ChromeDriver was started successfully.
[38;5;128m[Chromedriver@fc0e][0m [STDERR] [1721300962.181][INFO]: /dev/shm not writable, adding --disable-dev-shm-usage switch
[38;5;128m[Chromedriver@fc0e][0m [STDERR] [1721300962.183][SEVERE]: bind() failed: Cannot assign requested address (99)
[38;5;128m[Chromedriver@fc0e][0m [STDERR] [1721300962.183][INFO]: listen on IPv6 failed with error ERR_ADDRESS_INVALID
[38;5;128m[Chromedriver@fc0e][0m Got response with status 200: {"value":{"build":{"version":"126.0.6478.182 (5b5d8292ddf182f8b2096fa665b473b6317906d5-refs/branch-heads/6478@{#1776})"},"message":"ChromeDriver ready for new sessions.","os":{"arch":"x86_64","name":"Linux","version":"5.10.218-208.862.amzn2.x86_64"},"ready":true}}
[38;5;128m[Chromedriver@fc0e][0m Starting W3C Chromedriver session with capabilities: {
[38;5;128m[Chromedriver@fc0e][0m   "capabilities": {
[38;5;128m[Chromedriver@fc0e][0m     "alwaysMatch": {
[38;5;128m[Chromedriver@fc0e][0m       "goog:chromeOptions": {
[38;5;128m[Chromedriver@fc0e][0m         "androidPackage": "com.test.stage",
[38;5;128m[Chromedriver@fc0e][0m         "androidUseRunningApp": true,
[38;5;128m[Chromedriver@fc0e][0m         "androidProcess": "com.test.stage",
[38;5;128m[Chromedriver@fc0e][0m         "androidDeviceSerial": "17291JEC207128"
[38;5;128m[Chromedriver@fc0e][0m       },
[38;5;128m[Chromedriver@fc0e][0m       "goog:loggingPrefs": {
[38;5;128m[Chromedriver@fc0e][0m         "browser": "ALL"
[38;5;128m[Chromedriver@fc0e][0m       }
[38;5;128m[Chromedriver@fc0e][0m     }
[38;5;128m[Chromedriver@fc0e][0m   }
[38;5;128m[Chromedriver@fc0e][0m }
[38;5;128m[Chromedriver@fc0e][0m Matched '/session' to command name 'createSession'
[38;5;128m[Chromedriver@fc0e][0m Proxying [POST /session] to [POST http://127.0.0.1:9515/session] with body: {"capabilities":{"alwaysMatch":{"goog:chromeOptions":{"androidPackage":"com.test.stage","androidUseRunningApp":true,"androidProcess":"com.test.stage","androidDeviceSerial":"17291JEC207128"},"goog:loggingPrefs":{"browser":"ALL"}}}}
[38;5;128m[Chromedriver@fc0e][0m [STDERR] [1721300962.190][INFO]: [9d77baac6e491c41b61d83a0ae545ff6] COMMAND InitSession {
[38;5;128m[Chromedriver@fc0e][0m [STDERR]    "capabilities": {
[38;5;128m[Chromedriver@fc0e][0m [STDERR]       "alwaysMatch": {
[38;5;128m[Chromedriver@fc0e][0m [STDERR]          "goog:chromeOptions": {
[38;5;128m[Chromedriver@fc0e][0m [STDERR]             "androidDeviceSerial": "17291JEC207128",
[38;5;128m[Chromedriver@fc0e][0m [STDERR]             "androidPackage": "com.test.stage",
[38;5;128m[Chromedriver@fc0e][0m [STDERR]             "androidProcess": "com.test.stage",
[38;5;128m[Chromedriver@fc0e][0m [STDERR]             "androidUseRunningApp": true
[38;5;128m[Chromedriver@fc0e][0m [STDERR]          },
[38;5;128m[Chromedriver@fc0e][0m [STDERR]          "goog:loggingPrefs": {
[38;5;128m[Chromedriver@fc0e][0m [STDERR]             "browser": "ALL"
[38;5;128m[Chromedriver@fc0e][0m [STDERR]          }
[38;5;128m[Chromedriver@fc0e][0m [STDERR]       }
[38;5;128m[Chromedriver@fc0e][0m [STDERR]    }
[38;5;128m[Chromedriver@fc0e][0m [STDERR] }
`

**Comment by mykola-mokhnach** [comment]
> i am using the following web driver manager https://mvnrepository.com/artifact/io.github.bonigarcia/webdrivermanager/5.8.0 to handle the chrome driver if the device is running uses old chrome version

I'm not sure how this is possible. The chromedriver binary should be located on the same host where Appium server is running. Thus it cannot work if a test is executed in a remote device farm.

The log above looks ok to me, I don't see any errors

**Comment by omarzero130** [comment]
i am using this and it works fine locally when I run it , do you recommend to add my chromedriver executable file within the project instead of using this ?

and for the log part the error starts from the following part 
` Sending adb command: host:transport:17291JEC207128|shell:grep -a '@webview_devtools_remote_.*4725' /proc/net/unix
[38;5;128m[Chromedriver@fc0e][0m [STDERR] [1721300962.374][DEBUG]: Received adb response: 0000000000000000: 00000002 00000000 00010000 0001 01 402031 @webview_devtools_remote_4725
[38;5;128m[Chromedriver@fc0e][0m [STDERR] [1721300962.374][DEBUG]: Sending adb command: host-serial:17291JEC207128:forward:tcp:0;localabstract:webview_devtools_remote_4725
[38;5;128m[Chromedriver@fc0e][0m [STDERR] [1721300962.382][DEBUG]: ADB server responded with "OKAYOKAY" instead of "OKAY".
[38;5;128m[Chromedriver@fc0e][0m [STDERR] [1721300962.382][DEBUG]: Received adb response: 40909
[38;5;128m[Chromedriver@fc0e][0m [STDERR] [1721300962.382][DEBUG]: DevTools HTTP Request: http://localhost:40909/json/version
[38;5;128m[Chromedriver@fc0e][0m [STDERR] [1721300962.383][DEBUG]: DevTools HTTP Request failed
[38;5;128m[Chromedriver@fc0e][0m [STDERR] [1721300962.433][DEBUG]: DevTools HTTP Request: http://localhost:40909/json/version
[38;5;128m[Chromedriver@fc0e][0m [STDERR] [1721300962.433][DEBUG]: DevTools HTTP Request failed
[38;5;128m[Chromedriver@fc0e][0m [STDERR] [1721300962.483][DEBUG]: DevTools HTTP Request: http://localhost:40909/json/version
[38;5;128m[Chromedriver@fc0e][0m [STDERR] [1721300962.484][DEBUG]: DevTools HTTP Request failed
[38;5;128m[Chromedriver@fc0e][0m [STDERR] [1721300962.534][DEBUG]: DevToo` 

And it keeps looping here until it timesout and then it displays this exception 

`
SKIPPED CONFIGURATION: @BeforeClass testCases.LoginPageTC.before
SKIPPED: testCases.LoginPageTC.TC1Login
org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Response code 500. Message: An unknown server-side error occurred while processing the command. Original error: A new session could not be created. Details: session not created
from chrome not reachable 
Host info: host: 'ip-10-1-129-37.us-west-2.compute.internal', ip: '10.1.129.37'
Build info: version: '4.22.0', revision: 'c5f3146703*'
System info: os.name: 'Linux', os.arch: 'amd64', os.version: '5.10.218-208.862.amzn2.x86_64', java.version: '17.0.2'
Driver info: io.appium.java_client.android.AndroidDriver
Command: [null, newSession {desiredCapabilities=Capabilities {appium:app: app.apk, appium:autoGrantPermissions: true, appium:autoWebview: true, appium:automationName: UIAutomator2, appium:chromedriverPort: 9515, appium:disableWindowAnimation: true, appium:platformVersion: 11, appium:showChromedriverLog: true, platformName: ANDROID}, capabilities=[{appium:automationName=UIAutomator2, appium:chromedriverPort=9515, appium:platformVersion=11, appium:app=app.apk, appium:disableWindowAnimation=true, appium:autoWebview=true, platformName=ANDROID, appium:autoGrantPermissions=true, appium:showChromedriverLog=true}]}]
Capabilities {appium:app: app.apk, appium:autoGrantPermissions: true, appium:autoWebview: true, appium:automationName: UIAutomator2, appium:chromedriverPort: 9515, appium:disableWindowAnimation: true, appium:platformVersion: 11, appium:showChromedriverLog: true, platformName: ANDROID}
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:114)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.appium.java_client.remote.AppiumProtocolHandshake.createSession(AppiumProtocolHandshake.java:116)
	at io.appium.java_client.remote.AppiumProtocolHandshake.createSession(AppiumProtocolHandshake.java:96)
	at io.appium.java_client.remote.AppiumCommandExecutor.createSession(AppiumCommandExecutor.java:176)
	at io.appium.java_client.remote.AppiumCommandExecutor.execute(AppiumCommandExecutor.java:237)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:518)
	at io.appium.java_client.AppiumDriver.startSession(AppiumDriver.java:268)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:161)
	at io.appium.java_client.AppiumDriver.<init>(AppiumDriver.java:90)
	at io.appium.java_client.AppiumDriver.<init>(AppiumDriver.java:102)
	at io.appium.java_client.android.AndroidDriver.<init>(AndroidDriver.java:109)
	at util.BaseTestClass.intialize(BaseTestClass.java:76)
	at util.BaseTestClass.prepration(BaseTestClass.java:48)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethodConsideringTimeout(MethodInvocationHelper.java:69)
	at org.testng.internal.invokers.ConfigInvoker.invokeConfigurationMethod(ConfigInvoker.java:361)
	at org.testng.internal.invokers.ConfigInvoker.invokeConfigurations(ConfigInvoker.java:296)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:376)
	at org.testng.SuiteRunner.run(SuiteRunner.java:330)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1256)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1176)
	at org.testng.TestNG.runSuites(TestNG.java:1099)
	at org.testng.TestNG.run(TestNG.java:1067)
	at org.testng.TestNG.privateMain(TestNG.java:1414)
	at org.testng.TestNG.main(TestNG.java:1378)`

**Comment by mykola-mokhnach** [comment]
> i am using this and it works fine locally when I run it , do you recommend to add my chromedriver executable file within the project instead of using this ?

Like I said there is no way to provide a custom chromedriver executable on a remote cloud host.

The log above is a chromedriver log. I don't think there is anything that could be done from the driver side there. Consider creating an issue report to Google. Also make sure the most recent chromedriver version is used there. Consider contacting the cloud provider customer support regarding how to configure it or if that is possible in general.

**Comment by omarzero130** [comment]
@mykola-mokhnach  I searched alot for way to intiate the chromedriver and use it but I can't find anything related to setup running  webview app on device farm using chromedriver .. do you any relatable link for this?  and thanks for  your replies BTW

**Comment by mykola-mokhnach** [comment]
I don't think so. Only the cloud provider themselves (e.g, their customer support) know their infrastructure the best.

**Comment by mykola-mokhnach** [comment]
closed as third party issue

**Comment by omarzero130** [comment]
I solved the issue by not passing any capabilities from my code ( pass the capabilities object empty) and use the AWS device farm and set in the test-spec.yml file and you can refer to this link [https://docs.aws.amazon.com/devicefarm/latest/developerguide/custom-test-environments.html](url) and follow the steps

**Comment by alexander-lazaris-sonder** [comment]
@omarzero130 I have run into this same issue recently and cannot wrap my head around it, seem to be running in circles.

I have bundled the chromedriver binary into my test package & from the chromedriver logs, get the same error you get (`chromedriver session cannot be created`)

For your solution, could you share an example as a reference? I don't see how not providing an empty `capabilities/options` solves this chromedriver problem.

**Comment by alexander-lazaris-sonder** [comment]
@omarzero130 (or anyone else who comes across this) - I've emptied my android options object and an android session is successfully created in appium.

I've followed https://repost.aws/questions/QUe0wDfhPuQOa2OZQVvA4OAQ/device-farm-mshop-automation-i-am-getting-error-related-to-chrome-browser-when-i-try-to-perform-after-webview-switch, yet it was not successful for me. I have both the bash script & the test spec changes applied however device farm fails to recognise the commands from the tests spec. No luck with that solution unfortunately.
